libs/common/src/test/java/org/opensearch/common/round/RoundableTests.java

```diff
@@ -8,19 +8,53 @@
 
 package org.opensearch.common.round;
 
+import org.opensearch.common.SuppressForbidden;
 import org.opensearch.test.OpenSearchTestCase;
+import org.junit.BeforeClass;
 
 public class RoundableTests extends OpenSearchTestCase {
 
-    public void testFloor() {
-        int size = randomIntBetween(1, 256);
-        long[] values = new long[size];
-        for (int i = 1; i < values.length; i++) {
-            values[i] = values[i - 1] + (randomNonNegativeLong() % 200) + 1;
+    @BeforeClass
+    @SuppressForbidden(reason = "Sets the feature flag system property to a random value")
+    public static void setupClass() {
+        System.setProperty("opensearch.experimental.feature.simd.rounding.enabled", String.valueOf(randomBoolean()));
+    }
+
+    public void testRoundingEmptyArray() {
+        Throwable throwable = assertThrows(IllegalArgumentException.class, () -> RoundableFactory.create(new long[0], 0));
+        assertEquals("at least one value must be present", throwable.getMessage());
+    }
+
+    public void testRoundingSmallArray() {
+        int size = randomIntBetween(1, 64);
+        long[] values = randomArrayOfSortedValues(size);
+        Roundable roundable = RoundableFactory.create(values, size);
+
+        assertEquals("BidirectionalLinearSearcher", roundable.getClass().getSimpleName());
+        assertRounding(roundable, values, size);
+    }
+
+    @SuppressForbidden(reason = "Reads a package-private static field using reflection as it's only available on Java 20 and above")
+    public void testRoundingLargeArray() {
+        int size = randomIntBetween(65, 256);
+        long[] values = randomArrayOfSortedValues(size);
+        Roundable roundable = RoundableFactory.create(values, size);
+
+        boolean isBtreeSearchSupported;
+        try {
+            // Not supported below Java 20.
+            isBtreeSearchSupported = RoundableFactory.class.getDeclaredField("IS_BTREE_SEARCH_SUPPORTED").getBoolean(null);
```

Comment:
Why do you need reflection here? Just mirror package private IS_BTREE_SEARCH_SUPPORTED to other `RoundableFactory` with value `false`

Reply:
I considered this approach too, but I felt that `IS_BTREE_SEARCH_SUPPORTED` is an implementation detail of the Java 20 source which shouldn't leak into the main source.

Reply:
Actually we don't know if the Vector API would be probed or not. Not only we need Java 20, we also want the platform to have a minimum of 4 long vector lanes for BtreeSearcher to be used.

The assertions currently ensure that the correct class between BtreeSearcher/BinarySearcher is supplied based on what's supported by the platform.

Reply:
I see, so there are 2 issues here: 
 1. we cannot reliable assert what the value of IS_BTREE_SEARCH_SUPPORTED should be (we could only check what it is but not why it is `true` or `false`) 
 2. since this is implementation detail (as you mentioned before), we could only assert that it is either `BtreeSearcher` or `BinarySearcher` without exposing IS_BTREE_SEARCH_SUPPORTED 

So basically it opens up 2 solutions: 
 - expose IS_BTREE_SEARCH_SUPPORTED (as package private)
 - assert `BtreeSearcher` or `BinarySearcher`

Reply:
I like the second approach. Since it's an implementation detail anyways, we can drop the reflective lookup, and simply check if the returned object is either `BtreeSearcher` or `BinarySearcher` as you suggested.